packages/datadog-plugin-google-cloud-pubsub/src/index.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+function createWrapRequest (tracer, config) {
+  return function wrapRequest (request) {
+    return function wrappedRequest (cfg = { reqOpts: {} }, cb) {
+      let topic
+      if (cfg.reqOpts) {
+        topic = cfg.reqOpts[cfg.method === 'createTopic' ? 'name' : 'topic']
+      }
+      const tags = {
+        component: '@google-cloud/pubsub',
+        'resource.name': `${cfg.method || 'request'}:${topic}`,
+        'service.name': config.service || `${tracer._service}-pubsub`,
+        'pubsub.method': cfg.method,
+        'pubsub.project_id': this.projectId,
+        'pubsub.topic': topic,
+        'span.kind': 'producer'
+      }
+      cb = tracer.scope().bind(cb)
+      return tracer.trace('pubsub.request', { tags }, (span, done) => {
+        if (cfg.reqOpts && cfg.method === 'publish') {
+          for (const msg of cfg.reqOpts.messages) {
+            if (!msg.attributes) {
+              msg.attributes = {}
+            }
+            tracer.inject(span, 'text_map', msg.attributes)
+          }
+        }
+
+        arguments[1] = function (err) {
+          done(err)
+          return cb.apply(this, arguments)
+        }
+
+        return request.apply(this, arguments)
+      })
+    }
+  }
+}
+
+function createWrapSubsciptionEmit (tracer, config) {
+  return function wrapSubscriptionEmit (emit) {
+    return function wrappedSubscriptionEmit (eventName, message) {
+      if (eventName !== 'message' || !message) {
+        return emit.apply(this, arguments)
+      }
+
+      const topic = this.metadata ? this.metadata.topic : 'default'
+      const tags = {
+        component: '@google-cloud/pubsub',
+        'resource.name': `receive:${topic}`,
```

Comment:
Since the topic should be undefined instead of having a default value, this should be combined with a join.